### ISSUE TITLE
Lab 8 Level 1 / CompletableFuture in general

### ISSUE THREAD

**Issue by nowknowing** [nus-cs2030/2021-s1#571]
# Description 
How do we catch the IO exception expected from the first task run asynchronously?
![image](https://user-images.githubusercontent.com/49021362/100579067-7337bc00-331e-11eb-8c3a-2485ca4e4c09.png)
![image](https://user-images.githubusercontent.com/49021362/100579100-8480c880-331e-11eb-927a-aed0d6dd0d61.png)

## Topic: 
Asynchronous

## Screenshots (if any): 
Insert Images here if necessary

**Comment by gabrielloye** [comment]
I don't think you need to catch the IOException/InterruptedException? The test cases on CodeCrunch seem to pass without handling these 2

**Comment by tjoewel** [comment]
More generally, CompletableFuture is a functor and a monad (lecture 12 pg 17). What the sendAsync method does is return a HttpResponse wrapped in a CompletableFuture. If you look at the java API for HttpClient, you will see that `send` throws some exceptions of which IOException is one of them. However, `SendAsync` only throws IllegalArgumentException. All other exceptions would be handled in CompletableFuture similar to how the Sandbox in lecture 7 handles exceptions. If there is an IOException, it would be thrown when you do CompletableFuture.join(), so simply assigning a variable a HttpResponse wrapped in a CompletableFuture will not throw a IOException. Hope this helps!
